src/KnownTypes.php

```diff
@@ -0,0 +1,66 @@
+ *     * Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following disclaimer
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *     * Neither the name of Google Inc. nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+namespace Google\ApiCore;
+
+class KnownTypes
+{
+    public const GRPC_TYPES = [
+        'google.rpc.retryinfo-bin' => \Google\Rpc\RetryInfo::class,
+        'google.rpc.debuginfo-bin' => \Google\Rpc\DebugInfo::class,
+        'google.rpc.quotafailure-bin' => \Google\Rpc\QuotaFailure::class,
+        'google.rpc.badrequest-bin' => \Google\Rpc\BadRequest::class,
+        'google.rpc.requestinfo-bin' => \Google\Rpc\RequestInfo::class,
+        'google.rpc.resourceinfo-bin' => \Google\Rpc\ResourceInfo::class,
+        'google.rpc.errorinfo-bin' => \Google\Rpc\ErrorInfo::class,
+        'google.rpc.help-bin' => \Google\Rpc\Help::class,
+        'google.rpc.localizedmessage-bin' => \Google\Rpc\LocalizedMessage::class,
+        'google.rpc.preconditionfailure-bin' => \Google\Rpc\PreconditionFailure::class,
+    ];
+
+    public const JSON_TYPES = [
+        'type.googleapis.com/google.rpc.RetryInfo' => \Google\Rpc\RetryInfo::class,
+        'type.googleapis.com/google.rpc.DebugInfo' => \Google\Rpc\DebugInfo::class,
+        'type.googleapis.com/google.rpc.QuotaFailure' => \Google\Rpc\QuotaFailure::class,
+        'type.googleapis.com/google.rpc.BadRequest' => \Google\Rpc\BadRequest::class,
+        'type.googleapis.com/google.rpc.RequestInfo' => \Google\Rpc\RequestInfo::class,
+        'type.googleapis.com/google.rpc.ResourceInfo' => \Google\Rpc\ResourceInfo::class,
+        'type.googleapis.com/google.rpc.ErrorInfo' => \Google\Rpc\ErrorInfo::class,
+        'type.googleapis.com/google.rpc.Help' => \Google\Rpc\Help::class,
+        'type.googleapis.com/google.rpc.LocalizedMessage' => \Google\Rpc\LocalizedMessage::class,
+        'type.googleapis.com/google.rpc.PreconditionFailure' => \Google\Rpc\PreconditionFailure::class,
```

Comment:
Can you handle a custom error detail type that is defined as part of the API but is not on this list? (That's what the internal [go/cloudsdk-gapic-error-details-handling-status](http://goto.google.com/cloudsdk-gapic-error-details-handling-status) is about.)

Work to support this can/should be in a follow-up PR, but it may affect how you handle the standard error details here.